src/helpers/generateEnumType.ts

```diff
@@ -6,10 +6,22 @@ import isValidTSIdentifier from "~/utils/isValidTSIdentifier";
 import { generateStringLiteralUnion } from "./generateStringLiteralUnion";
 import { generateTypedReferenceNode } from "./generateTypedReferenceNode";
 
-export const generateEnumType = (name: string, values: DMMF.EnumValue[]) => {
+export type EnumType = {
+  objectDeclaration: ts.VariableStatement;
+  typeDeclaration: ts.TypeAliasDeclaration;
+  schema?: string;
+  typeName: string;
+};
+
+export const generateEnumType = (
+  name: string,
+  values: readonly DMMF.EnumValue[]
+): EnumType | undefined => {
   const type = generateStringLiteralUnion(values.map((v) => v.name));
 
-  if (!type) return [];
+  if (!type) {
+    return undefined;
```

Comment:
Is it more-correct to throw in this case? Or a better question to ask is "do we ned make sure an error is surfaced to the user here?"

I ask because there's code that uses generateEnumType (which can return undedfined) that does a check for `!!value` instead of potentially showing a warning/erroring

Reply:
The fact that `generateStringLiteralUnion` *could* return null means we should just surface the error in the case we do reach that condition, for whatever possible reason.  I think it's fine to just throw here

Reply:
Why break their entire schema generation when we can just skip the empty enum?